### PR TITLE
Five lavaland ruins

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
And associated code
- [x] Greed
- [x] Gluttony
- [x] Pride
- [x] Sloth
- [x] Envy

Other two will be done later but I want to refactor toggleable weapons first, so this is good to merge as is.

### CODE REVIEW

**Comment by RemieRichards** [comment]
Pictures.

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 1]
```diff
@@ -25,3 +25,74 @@
...
+		else
+			H << "You're not gluttonous enough to pass this barrier!"
+	else
+		return 0
+
+//Pride
+
+/obj/structure/mirror/magic
+	name = "pride's mirror"
+	desc = "Pride cometh before the..."
```
Comment (RemieRichards) [inline code/modules/ruins/lavaland_ruin_code.dm]: VERY clever.

**Comment by KorPhaeron** [comment]
@RemieRichards 

Spoiling the surprise!

http://i.imgur.com/IiMIqf2.png

**Comment by RemieRichards** [comment]
Neato.

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 2]
```diff
@@ -25,3 +25,74 @@
...
+	user << "You feel your very life draining away as you pull the lever...it'll be worth it though, right?"
+	user.adjustCloneLoss(20)
+	if(user.stat)
+		user.gib()
+	icon_state = "slots2"
+	sleep(5)
+	icon_state = "slots1"
+	in_use = FALSE
+	if(prob(win_prob))
+		new /obj/item/weapon/dice/d20/fate/one_use(get_turf(src))
```
Comment (duncathan) [inline code/modules/ruins/lavaland_ruin_code.dm]: dude that's hardcore

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 3]
```diff
@@ -25,3 +25,74 @@
...
+			user << "You hear laughter echoing around you as the machine fades away. In it's place...more gambling."
+			qdel(src)
+	else
+		if(user)
+			user << "Looks like you didn't win anything this time...next time though, right?"
+//Gluttony
+
+/obj/effect/gluttony
+	name = "gluttony's wall"
+	desc = "Only those who truely indulge may pass."
```
Comment (duncathan) [inline code/modules/ruins/lavaland_ruin_code.dm]: truly

**Comment by duncathan** [comment]
Loving the deadly sins theme

File: code/modules/reagents/reagent_containers/syringes.dm [hunk 4]
```diff
@@ -246,4 +246,11 @@
 	desc = "A syringe used to completely change the users identity."
 	amount_per_transfer_from_this = 1
 	volume = 1
-	list_reagents = list("mulligan" = 1)
\ No newline at end of file
+	list_reagents = list("mulligan" = 1)
+
+/obj/item/weapon/reagent_containers/syringe/gluttony
+	name = "Gluttony's Blesssing"
```
Comment (Cheridan) [inline code/modules/reagents/reagent_containers/syringes.dm]: KOR CONFIRMED FOR LIZARDPERSON 
SHUT DOWN EVERYTHING

THIS IS NOT A DRILL

**Comment by Jazaen** [comment]
Lavaland is happening! Slowly, but happening!
What are the main themes of each ruin except naming? Do you have to commit one of the 7 sins in each temple?

File: _maps/templates/lavaland_surface_sloth.dmm [hunk 5]
```diff
@@ -0,0 +1,20 @@
+"a" = (/turf/indestructible/riveted{baseturf = /turf/simulated/floor/plating/lava/smooth},/area/ruin/unpowered)
+"b" = (/turf/simulated/floor/plating/lava/smooth/lava_land_surface,/area/ruin/unpowered)
+"c" = (/obj/item/weapon/paper{desc = "have not gotten around to finishing my cursed item yet sorry - sloth"; name = "note from sloth"},/turf/simulated/floor/sepia{baseturf = /turf/simulated/floor/plating/lava/smooth; blocks_air = 1; slowdown = 10},/area/ruin/unpowered)
```
Comment (RemieRichards) [inline _maps/templates/lavaland_surface_sloth.dmm]: *slow clap*

**Comment by KorPhaeron** [comment]
@Jazaen 

You will have to explore them and find out!

Or read the code whichever

**Comment by CosmicScientist** [comment]
:+1:*:100: _awaiting addition of betrayal, the deadliest of sins_

**Comment by KazeEspada** [comment]
Lust should be lizard themed.

**Comment by Supermichael777** [comment]
Don't make lust a injector that turns you into a xeno. That needs to STAY a meme.
As for Wrath a black ATV that lets you smash through shit and tells you to run people down would be cool.  ROAD RAGE RAMPAGE.

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 6]
```diff
@@ -25,3 +25,104 @@
...
+/obj/structure/cursed_slot_machine/attack_hand(mob/living/carbon/human/user)
+	if(!istype(user))
+		return
+	if(in_use)
+		return
+	if(!ishuman(user))
+		return
+	in_use = TRUE
+	user << "You feel your very life draining away as you pull the lever...it'll be worth it though, right?"
+	user.adjustCloneLoss(20)
```
Comment (Iamgoofball) [inline code/modules/ruins/lavaland_ruin_code.dm]: does cloneloss even factor into health?

Comment (KazeEspada) [inline code/modules/ruins/lavaland_ruin_code.dm]: It does? Have you been cloned recently?

File: code/modules/awaymissions/mission_code/Academy.dm [hunk 7]
```diff
@@ -167,6 +167,9 @@
 	var/used = 0
 	var/rigged = -1
 
+/obj/item/weapon/dice/d20/fate/one_use
+	reusable = 0
```
Comment (Zilenan91) [inline code/modules/awaymissions/mission_code/Academy.dm]: Stealth dice of fate nerf?

Comment (RemieRichards) [inline code/modules/awaymissions/mission_code/Academy.dm]: @Zilenan91 no

Comment (PKPenguin321) [inline code/modules/awaymissions/mission_code/Academy.dm]: it's a child item (AKA an item that inherits all of the parent item's traits, and can be tweaked without altering the parent item) that is one-use

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 8]
```diff
@@ -25,3 +25,104 @@
...
+	icon_state = "magic_mirror"
+
+/obj/structure/mirror/magic/pride/curse(mob/user)
+	user.visible_message("<span class='danger'><B>The ground splits beneath [user] as their hand leaves the mirror!</B></span>")
+	var/turf/T = get_turf(user)
+	T.ChangeTurf(/turf/simulated/chasm/straight_down)
+	var/turf/simulated/chasm/straight_down/C = T
+	C.drop(user)
+
+//Sloth - I'll finish this item later
```
Comment (duncathan) [inline code/modules/ruins/lavaland_ruin_code.dm]: :+1:

**Comment by Thunder12345** [comment]
\>First comment asks for pictures
\>Still no pictures
smh tbh fam

**Comment by RemieRichards** [comment]
@Thunder12345 the pictures were posted, and later on removed.
because obvi they're spoilery.

File: _maps/templates/lavaland_surface_sloth.dmm (hunk 5 above)
Comment (AnturK) [inline _maps/templates/lavaland_surface_sloth.dmm]: Hilarious

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 9]
```diff
@@ -25,3 +25,104 @@
...
+	anchored = 1
+	density = 1
+	var/win_prob = 5
+
+/obj/structure/cursed_slot_machine/attack_hand(mob/living/carbon/human/user)
+	if(!istype(user))
+		return
+	if(in_use)
+		return
+	if(!ishuman(user))
```
Comment (AnturK) [inline code/modules/ruins/lavaland_ruin_code.dm]: You already checked the type once ?

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 10]
```diff
@@ -25,3 +25,104 @@
...
+//Pride
+
+/obj/structure/mirror/magic
+	name = "pride's mirror"
+	desc = "Pride cometh before the..."
+	icon_state = "magic_mirror"
+
+/obj/structure/mirror/magic/pride/curse(mob/user)
+	user.visible_message("<span class='danger'><B>The ground splits beneath [user] as their hand leaves the mirror!</B></span>")
+	var/turf/T = get_turf(user)
```
Comment (AnturK) [inline code/modules/ruins/lavaland_ruin_code.dm]: This is more mirror base code related but does it check for adjacency after the choice ? So you can't keep it open for later use.

File: code/modules/ruins/lavaland_ruin_code.dm [hunk 11]
```diff
@@ -25,3 +25,104 @@
...
+		if(H.nutrition >= NUTRITION_LEVEL_FAT)
+			return 1
+		else
+			H << "<span class='danger'><B>You're not gluttonous enough to pass this barrier!</B></span>"
+	else
+		return 0
+
+//Pride
+
+/obj/structure/mirror/magic
```
Comment (KazeEspada) [inline code/modules/ruins/lavaland_ruin_code.dm]: How did all the maintainer's miss this? You overrode the parent object. This should be /obj/structure/mirror/magic/pride

File: code/modules/ruins/lavaland_ruin_code.dm (hunk 10 above)
Comment (KorPhaeron) [inline code/modules/ruins/lavaland_ruin_code.dm]: Fixed it in the mirror file, thanks

**Pull request merged**